pkg/cluster/deploystorage_test.go

```diff
@@ -81,3 +91,82 @@ func TestDenyAssignments(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateAndUpdateErrors(t *testing.T) {
+	ctx := context.Background()
+	clusterID := "test-cluster"
+	resourceGroupName := "fakeResourceGroup"
+	resourceGroup := fmt.Sprintf("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/%s", resourceGroupName)
+	location := "eastus"
+
+	group := mgmtfeatures.ResourceGroup{
+		Location:  &location,
+		ManagedBy: &clusterID,
+	}
+
+	disallowedByPolicy := autorest.NewErrorWithError(&azure.RequestError{
+		ServiceError: &azure.ServiceError{Code: "RequestDisallowedByPolicy"},
+	}, "", "", nil, "")
+
+	for _, tt := range []struct {
+		name    string
+		result  mgmtfeatures.ResourceGroup
+		mocks   func(*mock_features.MockResourceGroupsClient, interface{})
+		wantErr string
+	}{
+		{
```

Comment:
nit: add a use case where no error is returned and the function works

Reply:
Done :)